spec/rubocop/cop/lint/unmodified_reduce_accumulator_spec.rb

```diff
@@ -373,15 +373,42 @@
         RUBY
       end
 
+      it 'does not register an offense when calling a method on the accumulator with the element' do
+        expect_no_offenses(<<~RUBY)
+          foo.#{method} { |result, key| result.method(key) }
+        RUBY
+      end
+
+      it 'registers an offense when returning an index of the accumulator' do
+        expect_offense(<<~RUBY)
+          %w(a b c).#{method}({}) do |acc, letter|
+            acc[foo]
+            ^^^^^^^^ Do not return an element of the accumulator in `#{method}`.
+          end
+        RUBY
+      end
+
       it 'registers an offense when returning an index setter on the accumulator' do
         expect_offense(<<~RUBY)
           %w(a b c).#{method}({}) do |acc, letter|
-            acc[letter] = true
-            ^^^^^^^^^^^^^^^^^^ Do not return an element of the accumulator in `#{method}`.
+            acc[foo] = bar
+            ^^^^^^^^^^^^^^ Do not return an element of the accumulator in `#{method}`.
           end
         RUBY
       end
 
+      it 'does not register an offense when returning accumulator[element]' do
+        expect_no_offenses(<<~RUBY)
+          foo.#{method} { |result, key| result[key] }
+        RUBY
+      end
+
+      it 'does not register an offense when returning accumulator[element]=' do
+        expect_no_offenses(<<~RUBY)
+          foo.#{method} { |result, key| result[key] = foo }
```

Comment:
Just curious, but is there a legitimate use case for this form?

Reply:
Probably not, other than it being of the form `result.send <anything>, key` as in the issue. I can remove it.

Reply:
Updated to remove the exception for `[]=`

Reply:
Ah ah, you got me, I mispoke 😆.
Thanks for updating it; maybe someone will come up with a good usecase but I can't think of one.

Reply:
> Ah ah, you got me, I mispoke 😆.

Well, I take that back, `result.send <anything>, key` doesn't actually fit `:[]=`; that requires an extra argument! My honor is safe 🤣

Reply:
Yeah I guess I interpreted it as `key, ...` but in any case all good now 😄